### PR TITLE
Rename and update sequence

### LINKED ISSUE #812: BUG-Add sequence in a Scenario Context Manager
**Description**
When adding a sequence in a scenario context, there is a `KeyError` exception raised:

```
File "......\taipy\core\scenario\scenario.py", line 213, in add_sequence
    if not self.sequences[name]._is_consistent():
   ~~~~~~~~~~~~~~^^^^^^
KeyError: 'Sequence2'
```

**How to reproduce**

- A code fragment
    ```
    with scenario:
            seqs = scenario.sequences
            scenario.add_sequence(f"Sequence{len(seqs) +1}", list(seqs["baseline"].tasks.values()))
    ```

**Expected behavior**
No Error

**Screenshots**
When available and relevant, screenshots to better highlight the problem.

**Runtime environment**
Please specify relevant indications.
 - Taipy version (or branch name): develop
 - OS: Windows 11
 - Browser: Chrome
 - Python 3.11.4

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Create related issue in taipy-doc for documentation and Release Notes if relevant

### PR DESCRIPTION
**Pull request by jrobinAV**
Resolves #812 

- Opened a new API at the scenario level to rename a sequence
- Opened a new API at the scenario level to update an existing sequence
- Updated the `add_sequence` method to check if a sequence already exists with the same name.
- Updated the `NonExistingSequence` exception message.
- Accept spaces in sequence names.

### CODE REVIEW

**Comment by FredLL-Avaiga** [comment]
>@FredLL-Avaiga , you should use these APIs without a context manager instead.

I would need the context manager to update the tasks list

File: taipy/core/scenario/scenario.py [hunk 1]
```diff
@@ -269,6 +303,27 @@ def remove_sequences(self, sequence_names: List[str]):
...
+        """Rename a sequence of the scenario.
+
+        Parameters:
+            old_name (str): The current name of the sequence to rename.
+            new_name (str): The new name of the sequence.
+
+        Raises:
+            SequenceAlreadyExists^: If a sequence with the same name already exists in the scenario.
+        """
+        if new_name in self.sequences:
```
Comment (FabienLelaquais) [inline taipy/core/scenario/scenario.py:325]: I would have first returned if old == new name

File: taipy/core/sequence/sequence.py [hunk 2]
```diff
@@ -74,7 +74,8 @@ def __init__(
 
     @staticmethod
     def _new_id(sequence_name: str, scenario_id) -> SequenceId:
-        return SequenceId(Sequence._SEPARATOR.join([Sequence._ID_PREFIX, _validate_id(sequence_name), scenario_id]))
+        seq_id = sequence_name.replace(" ", "_")
```
Comment (FabienLelaquais) [inline taipy/core/sequence/sequence.py:77]: Sure about that? Only spaces are a problem?

**Review by FabienLelaquais — DISMISSED** [review]
Looking good

**Review by FredLL-Avaiga — DISMISSED** [review]

File: tests/core/scenario/test_scenario.py [hunk 3]
```diff
@@ -345,8 +344,133 @@ def test_raise_tasks_not_in_scenario_with_add_sequence_api(data_node):
...
+    assert scenario.sequences["seq_1"].tasks == {"foo": task_1}
+    assert scenario.sequences["seq_1"].name == "seq_1"
+    scenario.update_sequence("seq_1", [task_2], {"new_key": "new_value"}, [])
+    assert len(scenario.sequences) == 1
+    assert scenario.sequences["seq_1"].tasks == {"bar": task_2}
+    assert scenario.sequences["seq_1"].name == "seq_1"
+    assert scenario.sequences["seq_1"].properties["new_key"] == "new_value"
+
+
+def test_add_rename_and_remove_sequences_within_context(data_node):
```
Comment (jrobinAV) [inline tests/core/scenario/test_scenario.py:449]: @FredLL-Avaiga This test exhibits the fix

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**